src/generate_counterfactual/growing_spheres.jl

```diff
@@ -8,11 +8,8 @@ function generate_counterfactual(
     num_counterfactuals::Int=1,
```

Comment:
Tests are failing because the existing core method expects additional keyword arguments. A simple way to deal with this is to just `, kwrgs...` at the end of the method signature:

```julia
function generate_counterfactual(
    x::AbstractArray,
    target::RawTargetType,
    data::DataPreprocessing.CounterfactualData,
    M::Models.AbstractFittedModel,
    generator::Generators.GrowingSpheresGenerator;
    num_counterfactuals::Int=1,
    ...,         # explicitly add any other kwrgs that are used here 
    kwrgs...,    # use splashing here to allow passing kwrgs that are not relevant
)
```

Reply:
>Tests are failing because the existing core method expects additional keyword arguments.

@pat-alt, something to be refactored as well. Additional kwargs should never be expected.

Reply:
Well I'm saying 'additional' because that's the currently the baseline. So when we overload the baseline method with a method that needs less keyword arguments, then we need `kwrgs...` to ensure that the overloaded method still accepts the default `kwrgs` even if not needed. That is pretty common and I don't think there's anything wrong with it, but if you can refactor to do without it, even better.

Reply:
I'm not sure I follow.

Here is a little test I ran:

```
julia> function yolo(x::Int; y=1, z=0) print(x) end
yolo (generic function with 1 method)

julia> yolo(2)
2
julia> function yolo(x::String; y=1) print(x) end
yolo (generic function with 2 methods)

julia> yolo("Hello, Patrick")
Hello, Patrick
```

Reply:
Well, on a second thought, if it is a convention that those `kwrgs...` are irrelevant for that particular dispatch, then this is not a problem at all. @pat-alt, looking for your experience here :)

Reply:
Yup, that's right. For benchmarking, we specify that convergence should depend on generator conditions and that is for good reason: if we converge as soon as the threshold is reached, then the penalties of many generators never take any effect, which makes it impossible to effectively evaluate them. So I would keep the test suite general, because this is a general rule for benchmarking. But happy to discuss more indeed!